userspace/libsinsp/test/filter_parser.ut.cpp

```diff
@@ -385,57 +383,66 @@ TEST(parser, parse_position_info)
 // complex test case with all supported node types
 TEST(parser, expr_all_node_types)
 {
+	std::vector<std::unique_ptr<expr>> and_children;
+	and_children.push_back(unary_check_expr::create("evt.name", "", "exists"));
+	and_children.push_back(binary_check_expr::create("evt.type", "", "in", list_expr::create({"a", "b"})));
+	and_children.push_back(not_expr::create(binary_check_expr::create("evt.dir", "", "=", value_expr::create("<"))));
+
+	std::vector<std::unique_ptr<expr>> or_children;
+	or_children.push_back(and_expr::create(and_children));
+	or_children.push_back(binary_check_expr::create("proc.name", "", "=", value_expr::create("cat")));
+
+	std::unique_ptr<expr> ast = or_expr::create(or_children);
```

Comment:
This is also less readable, but the simplest way I found to pass a `std::vector<std::unique_ptr>>` around to be moved seems to be via lvalue reference, so it needs a named object